consensus/enclave/impl/src/lib.rs

```diff
@@ -108,6 +108,13 @@ pub struct SgxConsensusEnclave {
     /// Logger.
     logger: Logger,
 
+    /// Block version
+    ///
+    /// This is the value we were configured with on startup, and it goes into
+    /// the responder id. The enclave only forms blocks using this block
+    /// version.
+    block_version: Mutex<BlockVersion>,
```

Comment:
I am feeling a bit uncomfortable with the growing amount of mutexes we are piling up here. Maybe we should put all of the configuration data that comes from initialization in a single struct and put that behind a mutex?

Reply:
If we put this struct inside `mc-consensus-enclave-api` then we wont have to extend the init api call every time we add more configuration (which involves changing the untrusted crate, two mock enclaves, this crate, the messages file, etc...)

Reply:
and we can Digestible it and use that for the responder-id :)